bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -71,9 +100,19 @@ public void propertyChange(PropertyChangeEvent evt) {
 			int percentVisible = (int) (100 * a.getMean(Circumstance.TIMING_WINDOW_OPEN, false));
 			switch (percentVisible) {
 			case 100: // good!
-				if (a.getObs().getTimingWindows().size() > 0)
+				if (a.getObs().getTimingWindows().size() > 0 && !night.isEmpty())
 					mm.addMarker(true, this, Severity.Info, "Timing constraint is met.", v, a);
-				break;
+
+                    // Report timing windows
+                    final TimingWindowSolver timingWindowSolver = new TimingWindowSolver(a.getObs());
+                    final ImList<Interval> timingWindowIntervals = DefaultImList.create(timingWindowSolver.solve(night.getValue()).getIntervals());
+
+                    if (!timingWindowIntervals.nonEmpty()) {
+						final String msg = timingWindowIntervals.map(i -> formatInterval(zone, i)).mkString("Must be in the timing window ", ", ", ".");
```

Comment:
`!timingWindowIntervals.nonEmpty()` == `timingWindowIntervals.isEmpty()` in which case the `msg` if added would always be "Must be in the timing window ."?  Am I reading it wrong?